### PR TITLE
adding initial changes to EDI 816 and 824 - add'l segments and the 2 typ...

### PR DESCRIPTION
**Pull request by taronson**
...es

EDI related classes

### CODE REVIEW

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/ReferenceIdentificationInvalidTxn.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package javax.edi.model.x12.segment;
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/ReferenceIdentificationInvalidTxn.java]: This segment already exist as ReferenceNumber.java

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/StatusIndicator.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
+package javax.edi.model.x12.segment;
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/StatusIndicator.java]: This segment already exist as BaselineItemData.java

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/NoteErrorMessage.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package javax.edi.model.x12.segment;
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/NoteErrorMessage.java:1]: Is this needed?

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupTrailer.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package javax.edi.model.x12.segment;
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupTrailer.java]: This segment already exist as GroupEnvelopeTrailer

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupHeader.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
+package javax.edi.model.x12.segment;
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupHeader.java]: This segment already exist as GroupEnvelopeHeader

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/ReferenceIdentificationInvalidTxn.java (hunk 1 above)
Comment (taronson) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/ReferenceIdentificationInvalidTxn.java]: Didn't think these were exactly the same. OK

----- Original Message -----

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/segment/ApplicationAdviceBody.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package javax.edi.model.x12.edi824.segment;
+
+import javax.edi.model.x12.edi824.segment.Detail;
+import javax.edi.model.x12.edi824.segment.Header;
+import javax.edi.model.x12.edi824.segment.Trailer;
+import javax.validation.constraints.NotNull;
+
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/segment/ApplicationAdviceBody.java:8]: Missing @EDISegmentGroup

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/ApplicationAdvice.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+import javax.edi.bind.annotations.EDICollectionType;
+import javax.edi.model.x12.edi816.segment.OrgRelationshipsBody;
+import javax.edi.model.x12.edi824.segment.ApplicationAdviceBody;
+import javax.edi.model.x12.segment.GroupEnvelopeHeader;
+import javax.edi.model.x12.segment.GroupEnvelopeTrailer;
+import javax.edi.model.x12.segment.InterchangeEnvelopeHeader;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/ApplicationAdvice.java:15]: Missing @EDISegmentGroup

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/NoteErrorMessage.java (hunk 3 above)
Comment (taronson) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/NoteErrorMessage.java:1]: I see it in specs and searched for class that was already there.

----- Original Message -----

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupHeader.java (hunk 5 above)
Comment (taronson) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/segment/FunctionalGroupHeader.java]: Same applies wrt exact match but maybe missed something.
----- Original Message -----

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/segment/ApplicationAdviceBody.java (hunk 6 above)
Comment (taronson) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/edi824/segment/ApplicationAdviceBody.java:8]: Good catch, thanks.

----- Original Message -----

File: jaeb-model-x12/src/main/java/javax/edi/model/x12/edi816/segment/OrgRelationshipsGroup.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+	@EDICollectionType(AddressInformation.class)
+	private Collection<AddressInformation> addrInfo;  // Party Location
+	
+	private GeographicLocation geoLocation;
+	private PersonContact contact;  // Admin. Comm. Contact
+	
+	@Size(min=0, max=10)
+	@EDICollectionType(DateTimeReference.class)
+	private Collection<DateTimeReference> date;
+
```
Comment (jeckste) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/edi816/segment/OrgRelationshipsGroup.java]: This needs to be the first field

Comment (taronson) [inline jaeb-model-x12/src/main/java/javax/edi/model/x12/edi816/segment/OrgRelationshipsGroup.java]: Like we talked about John, HL was first segment. Not sure why that version is out there.

----- Original Message -----

**Pull request merged**